File: src/main/resources/output/IFC4X3_vbnet/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcBuiltElement/IfcDoor/IfcDoor.vb
Public Class IfcDoor Inherits IfcBuiltElement
    Public OverallHeight As IfcPositiveLengthMeasure
    Public OverallWidth As IfcPositiveLengthMeasure
    Public PredefinedType As IfcDoorTypeEnum
    Public OperationType As IfcDoorTypeOperationEnum
    Public UserDefinedOperationType As IfcLabel

    ' === WHERE CLAUSES ===
    ' CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcDoorTypeEnum.USERDEFINED) OR ((PredefinedType = IfcDoorTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
    ' CorrectTypeAssigned : (SIZEOF(IsTypedBy) = 0) OR ('IFC4X3_DEV_73740fe4.IFCDOORTYPE' IN TYPEOF(SELF\IfcObject.IsTypedBy[1].RelatingType))
End Class

File: src/main/resources/output/IFC4X3_java/Entities/IfcRoot/IfcObjectDefinition/IfcObject/IfcProduct/IfcElement/IfcBuiltElement/IfcDoor/IfcDoor.java
public class IfcDoor extends IfcBuiltElement {
    public IfcPositiveLengthMeasure OverallHeight;
    public IfcPositiveLengthMeasure OverallWidth;
    public IfcDoorTypeEnum PredefinedType;
    public IfcDoorTypeOperationEnum OperationType;
    public IfcLabel UserDefinedOperationType;

    // === WHERE CLAUSES ===
    // CorrectPredefinedType : NOT(EXISTS(PredefinedType)) OR (PredefinedType <> IfcDoorTypeEnum.USERDEFINED) OR ((PredefinedType = IfcDoorTypeEnum.USERDEFINED) AND EXISTS (SELF\IfcObject.ObjectType))
    // CorrectTypeAssigned : (SIZEOF(IsTypedBy) = 0) OR ('IFC4X3_DEV_73740fe4.IFCDOORTYPE' IN TYPEOF(SELF\IfcObject.IsTypedBy[1].RelatingType))
}
